react/features/toolbox/components/web/ToolbarButton.js

```diff
@@ -41,6 +41,68 @@ class ToolbarButton extends AbstractToolbarButton<Props> {
         tooltipPosition: 'top'
     };
 
+    /**
+     * Initializes a new {@code ToolbarButton} instance.
+     *
+     * @inheritdoc
+     */
+    constructor(props: Props) {
+        super(props);
+
+        this._onKeyDown = this._onKeyDown.bind(this);
+        this._onKeyUp = this._onKeyUp.bind(this);
+    }
+
+    _onKeyDown: (Object) => void;
+
+    /**
+     * Handles 'Enter' key on the button to trigger onClick for accessibility.
+     *
+     * @param {Object} event - The key event.
+     * @private
+     * @returns {void}
+     */
+    _onKeyDown(event) {
+        // If the event coming to the dialog has been subject to preventDefault
+        // we don't handle it here.
+        if (event.defaultPrevented) {
+            return;
+        }
+
+        if (event.key === 'Enter') {
+            event.preventDefault();
+            event.stopPropagation();
+            this.props.onClick();
+        } else if (event.key === ' ') {
+            // Space triggers button onKeyUp but we need to prevent default here
+            event.preventDefault();
+        }
+    }
+
+    _onKeyUp: (Object) => void;
+
+    /**
+     * Handles ' '(Space) key on the button to trigger onClick for
+     * accessibility.
+     *
+     * @param {Object} event - The key event.
+     * @private
+     * @returns {void}
+     */
+    _onKeyUp(event) {
```

Comment:
Kind of a big code duplication between here and the other changed file for `_onKeyUp` and `_onKeyDown`…

Reply:
True, but getting rid of it would require a bigger refactor and it could mean this PR never lands. Ideally this accessibility handling part would be in some abstract button class.

Reply:
They can both vanish if the buttons use `<button />`
For https://github.com/jitsi/jitsi-meet/pull/5432/files#diff-c440f22c27ccfb17f363716e5702d175R116 at least that'd be easier, just the native browser button styles would need inhibiting.

Reply:
well… removing the default browser style is easy. And for semantic reasons, that should use the `button` tag anyway… :+1:

Reply:
so if it would be a `button` tag instead of `div` then the key up and down logics would not be necessary?

Reply:
Indeed, they are just emulating button behaviour